### ISSUE TITLE
env.ActionMaker and env.ActionBuilder not defined

### ISSUE THREAD

**Issue by aryamccarthy** [TomLippincott/steamroller#1]
In the latest release, I'm not able to access `ActionMaker` and `ActionBuilder` through the `env`. Also, as a consequence, `ActionBuilder` does not work when called directly; it relies on the inaccessible `env.ActionMaker`. This may be tied to the most recent edits that removed some early lines from `generate`. Then again, I might be missing something completely. Have I misunderstood the calling convention?

**Minimal example:**

```python
from SCons.Script import Environment, Variables
import steamroller


env = Environment(tools=[steamroller.generate])

env.Append(
    BUILDERS={
        "Split" : env.Builder(
            **env.ActionMaker(
                "python",
                "src/split.py",
                args="${SOURCES[0]} ${TARGETS[0]} ${TARGETS[1]}",
            )
        )
    }
)

data = env.File("data/my_data.txt.gz")
train, test = env.Split(
    ["work/train_${SPLIT}.txt", "work/test_${SPLIT}.txt"], data, SPLIT=1)
```

**Traceback:**

```
Traceback (most recent call last):
  File "/Users/arya/PycharmProjects/kilo/steamroller_test/SConstruct", line 10, in <module>
    **env.ActionMaker(
AttributeError: 'SConsEnvironment' object has no attribute 'ActionMaker'
```

**Comment by TomLippincott** [comment]
Nope, you're not missing anything, this was just me screwing around with a way to not require *any* non-vanilla-SCons, such that simply adding `steamroller.generate` to the `tools` argument of the environment constructor would convert build rules to be grid-aware.  I got it kinda-working, and then got sidetracked.

I'll definitely get around to cleaning this up over the Summer (there's also a lot of hard-coded and Univa-specific stuff in it, while Slurm is running on all the new JHU infrastructure).  But for the moment (and possibly much longer) I'd look at its state as "works for some major grid management config, so the distance to some other config is probably rather small".
